### DISCUSSION TITLE
Genmon latest Version V1.19.05. Email IP address is incorrece; older one, and battery charging current is not on the graph.

### DISCUSSION THREAD

**Discussion by daved44** [jgyates/genmon#1225] [General]
Generac 20K, Model 0062500; Serial 8097262, Evolution. Latest Genmon version email reports an incorrect, once-used IP address. The IP address reported was assigned with a previous router that I replaced several weeks ago. Can this be solved with a reset or manual reassignment of some sort? Plus, the IP address is not included on all emails. 
Additionally, battery charge current is not displayed as per the release notes.
Thanks for a great product!

**Comment by jgyates** [comment]
Hi @daved44,

The code that gets the IP address to display is here:

https://github.com/jgyates/genmon/blob/584a8052ead9ee20324d5d2a8356e76f5e67ecfc/genmonlib/mysupport.py#L528C9-L528C21

This opens a socket and connects to an IP address then displays the IP address of the connection. I am guessing you still have this IP address (possibly multiple IP addresses) assigned.

depending on your linux version you can dispaly the currently used IP addresses with one of these two commands


      ifconfig

or 

     ip a

You can try rebooting but it dependson on how the IP was assigned. If it is a static defined IP defined locally then you would need to remove the setting on your pi's network config. If it is a static defined but assigned by the router a reboot of the pi may correct.

Regarding the IP not included in all emails. I would need more info. There is an email sent when genmon starts. The IP address of the genmon system is the main part of the email body. For other status emails, the IP address is the last line of the email. If you are not seeing this in some email then please include your email in this thread.

The battery current is displayed as text (not a gauge) on the status page of the web interface. It is only displayed for Evolution 1.0 and 2.0. Synergy and Power Pact Controllers are not displayed. If you are not seeing this an have a supported controller you can send your log files from the about page and I can take a look.

**Reply by daved44** [reply]
I had missed the battery current, looking for a gauge. Thanks.

The connection is ethernet; the current IP address is 192.168.50.70.  fconfig and ip a both display the correct current address, as does the GUI for the Genmon Pi, a 2GB Pi 4.  Debian Bookworm OS. There is a WiFi connection defined, but it is disabled since I installed Genmon a year ago. However, I still have two connection names in  the GUI Wireless Editing tab (2.4G and 5G) that point to the old Wireless connection SSIDs, both DHCP, and no IP displayed. Could be the problem? 

I would delete the wireless connection but the GUI tab doesn't allow me to. If you can point me to a terminal command that would do it, that might sew it up. I just rebooted the Pi and both emails, Generator Monitor Starting  . . .  and Generator Notice:  . . .  still report the incorrect IP address. The weekly exercise will run tomorrow morning; I get four emails from that and have time today to make more changes, so will know then (if those emails report the IP address) whether these might fix it. 

As for the emails that report or don't report the IP address (all incorrect)
The first after update reboot:
![image](https://github.com/user-attachments/assets/2d79e0eb-8599-471f-97a2-5f9594102360)
The second, right after the first, does report the IP address at the bottom, many lines down:
![image](https://github.com/user-attachments/assets/df5d658e-2f8e-4cd5-8f1e-a2ddee2d1770)
Several test emails followed; none reported the IP address:
![image](https://github.com/user-attachments/assets/a399d6ea-01af-40a7-9ed9-827d92f38f95)

I'm not sure why that is important and it really makes on difference to me, but doesn't seem to be working properly.

**Reply by jgyates** [reply]
The test emails by design do not send any information. They are only test emails so this behavior is expected.

What is the output of this command:

     ip r

this shows the default route for your system. My guess is that this will show the IP address that genmon is showing.

Probably the simplest thing to do you your case would be to disable the wireless interface since you are using Ethernet. Your pi can have multiple network interfaces and multiple IP addresses. The default route (e.g. the network path tried first) is set in the system routing table. The default route can be changed but if you are not using the wireless then it is easier to just disable the wifi. You can remove the SSID using the UI, or you can use the raspi-config program to do so (type "sudo raspi-config" then select System Options then Wireless LAN and remove your SSID. To completely disable wifi you can edit the file /boot/config.txt (could be named /config.txt depending on how old your setup is) and add the line dtoverlay=disable-wifi to the bottom of the file.

In short, what you are seeing is expected, you just have multiple IP addresses and your pi is defaulting to the older one.  If you disable the old one the newer one will be used. Based on the way your pi is setup now, unless you connect explicitly to the ethernet connection (like specifying an IP address in your browser), your pi is going to use the wireless connection first for outbound connections.

**Reply by daved44** [reply]
ip r command output:

![image](https://github.com/user-attachments/assets/f29aef73-d2b2-47ad-ba12-a4a3d9223c14)

Using raspi-config, entering the SSID and the passphrase, for both the 5G and 2.4G (defined as separate SSIDs on my system), it fails with: 
![image](https://github.com/user-attachments/assets/a89ba87f-3af3-41a9-8892-01b7b18ca85c)

This is far as I am able to go today.

**Reply by jgyates** [reply]
so what happens if you try to connect via ssh to the IP address that genmon is showing?

Also, what is the output of this command:

    route


you may need to edit the file /etc/wap_supplicant/wpa_supplicant.conf and remove your SSID and passwords, then reboot to fully get rid of the SSID entries.

**Reply by daved44** [reply]
The problem is solved. I did execute the route command, and it showed all the current IP addresses, no problem. 

An explanation of the solution: When I tried using SSH to access the host from the _old IP address_ at your request, it just hung, as one would expect. But that got me to thinking about  when I was using SSH from the Windows desktop a year or more ago, at a time when I was learning Linux on the Pi. I had altered the windows hosts file to easily type ssh user@host instead of ssh user@ip address. The Windows host file, of course, had nothing to do with the problem, which was internal to the genmon Linux host, and I have been using it for years in the Windows environment.

But that got me to thinking about the Linux /etc/hosts file (which I had apparently learned about the existence of back then) and I took a look at it. It had the old IP address coupled to both the genmon host name and the localhost. I changed it to the current IP address and rebooted. The weekly exercise just ran and the generator notice emails show the correct IP address, as did the reboot notices.

I do not recall why I would ever have known about the Linux hosts file, nor why I would have placed entries in it, and online research does not help me on this. It is fairly obvious that it wasn't automatic via some process and that I must have been instructed to put the entries in, since I also updated it (later?) with an entry for one of my other Pi's so I could easily SSH from one Pi to the other (again, part of my experimenting and learning of Linux).

Can you or someone help me with a reason that I would have been sent to /etc/hosts to make these changes? Might it also be useful to let the user community know about this potential problem?

As usual, thank you for all this help.

**Reply by skipfire** [reply]
I've never seen someone use the /etc/hosts file on the device to include its own IP address, typically it just has your loopback entries like this:

```
127.0.0.1       localhost
::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters

127.0.1.1       devicename
```

I also looked through your post history on GitHub and you only appear to have 2 past ones and neither had any mention of /etc/hosts, perhaps another forum where you were seeking general Linux/Pi help?

**Reply by jgyates** [reply]
thanks for letting us know the source of the problem. I agree with @skipfire , I have never seen the local IP in the hosts file.

**Reply by daved44** [reply]
I still have no clue about why that was added; there is no record of that in my extensive genmon install notes. My other two Pi's, a Pi 4 8GB  I use for learning and experimenting and a Pi 5 8G I use to serve music streaming, haven't been touched and look just like Jon's example. I do have a note about adding the PlayPi entry, but that was for SSH from the genmon Pi to the PlayPi.

If I ever figure out why, I'll reopen this. Thanks again for the assistance!
